InstaDogFactory.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  InstaDogFactory.swift
+//  AC3.2-InstaCats-2
+//
+//  Created by Annie Tung on 10/23/16.
+//  Copyright © 2016 C4Q. All rights reserved.
+//
+
+import UIKit
+
+class InstaDogFactory {
+    static let manager: InstaDogFactory = InstaDogFactory()
+    private init() {}
+    
+    class func makeInstaDogs(fileName: String) -> [InstaDog]? {
+        
+        guard let instaDogsURL: URL = InstaDogFactory.manager.getResouceURL(from: fileName),
+            let instaDogData: Data = InstaDogFactory.manager.getData(from: instaDogsURL),
+            let instaDogsAll: [InstaDog] = InstaDogFactory.manager.getInstaDogs(from: instaDogData) else { return nil }
+        return instaDogsAll
+    }
+    
+    private func getResouceURL(from fileName: String) -> URL? {
+        guard let dotRange = fileName.rangeOfCharacter(from: CharacterSet.init(charactersIn: ".")) else { return nil }
+        
+        let fileNameComponent: String = fileName.substring(to: dotRange.lowerBound)
+        let fileExtensionComponent: String = fileName.substring(from: dotRange.upperBound)
+        let fileURL: URL? = Bundle.main.url(forResource: fileNameComponent, withExtension: fileExtensionComponent)
+        return fileURL
+    }
+    
+    private func getData(from url: URL) -> Data? {
+        let fileData: Data? = try? Data(contentsOf: url)
+        return fileData
+    }
+    
+    internal func getInstaDogs(from jsonData: Data) -> [InstaDog]? {
+        
+        do {
+            let instaDogJSONData: Any = try JSONSerialization.jsonObject(with: jsonData, options: [])
+            guard let instaDogJSONCasted: [String : AnyObject] = instaDogJSONData as? [String: AnyObject],
+                let instaDogArray: [AnyObject] = instaDogJSONCasted["dogs"] as? [AnyObject] else { return nil }
+            
+            var instaDogs: [InstaDog] = []
+            instaDogArray.forEach({ instaDogObject in
+                guard let instaDogName: String =
+                        instaDogObject["name"] as? String,
```

Comment:
Make sure the keys from the JSON are exactly the same when you type it in because even one misspelling can make your entire guard statement fall through to the else condition and therefore you're unable to pull down the dogs data. Does that make sense? I'm not sure which one is incorrect but in general be careful during parsing this data.